salt/states/file.py

```diff
@@ -2634,9 +2636,17 @@ def managed(name,
             contents = ''
             for part in validated_contents:
                 for line in part.splitlines():
-                    contents += line.rstrip('\n').rstrip('\r') + os.linesep
-            if contents_newline and not contents.endswith(os.linesep):
-                contents += os.linesep
+                    if not line == validated_contents[-1]:
```

Comment:
Some logical error here. Probably correct one is:
```python
if not line == part[-1]:
```
But it would be also wrong to compare the contents because the content could contains equal data, say
```
line
line
line
```
So you probably will want something like this:
```python
lines = part.splitlines()
for num, line in enumerate(lines):
    if not num == len(lines) - 1
```